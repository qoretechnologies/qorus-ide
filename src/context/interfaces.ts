import { createContext } from 'use-context-selector';
import {
  IQorusInterface,
  TQorusInterfaceCount,
} from '../containers/InterfacesView';

export interface IInterfacesContext {
  categories?: TQorusInterfaceCount;
  interfaces?: Record<string, IQorusInterface[]>;
  clone?: (type: string, id: string | number) => void;
  storage?: Record<string, any>;
  getStorage?: <T>(path: string, defaultValue?: T) => T;
  updateStorage?: <T>(path: string, value: T) => void;
  toggleEnabled?: (type: string, id: string | number, enable?: boolean) => void;
}

export const InterfacesContext = createContext<IInterfacesContext>({
  categories: {},
  interfaces: {},
  storage: {},
});
