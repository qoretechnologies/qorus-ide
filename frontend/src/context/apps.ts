import { createContext } from 'react';
import { IApp } from '../components/AppCatalogue';

export type TAppsContext = IApp[];

export const AppsContext = createContext<TAppsContext>([]);
