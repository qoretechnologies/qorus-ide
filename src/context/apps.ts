import { createContext } from 'react';
import { IApp } from '../components/AppCatalogue';
import { IActionSet } from '../containers/InterfaceCreator/fsm/ActionSetDialog';
import { IActionsSetsHookFunctions } from '../hooks/useActionSets';

export type TAppsContext = {
  apps?: IApp[];
  actionSets?: IActionSet[];
  loading?: boolean;
  error?: any;
  retry?: () => void;
} & IActionsSetsHookFunctions;

export const AppsContext = createContext<TAppsContext>({
  apps: [],
  actionSets: [],
});
