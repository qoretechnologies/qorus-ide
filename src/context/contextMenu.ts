import React, { createContext } from 'react';

import { TReqoreIntent } from '@qoretechnologies/reqore/dist/constants/theme';
import { IReqoreIconName } from '@qoretechnologies/reqore/dist/types/icons';

export interface IContextMenu {
  event: React.MouseEvent<HTMLElement>;
  data: {
    item?: string;
    onClick?: () => any;
    icon?: IReqoreIconName;
    rightIcon?: IReqoreIconName;
    title?: string;
    intent?: TReqoreIntent;
    disabled?: boolean;
  }[];
  onClose?: () => any;
}

export interface IContextMenuContext {
  addMenu?: (data: IContextMenu) => void;
  removeMenu?: () => void;
}

export const ContextMenuContext = createContext<IContextMenuContext>({});
