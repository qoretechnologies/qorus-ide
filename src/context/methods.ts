import { createContext } from 'react';

export interface IMethodsContext {
  showMethods?: boolean;
  setShowMethods?: any;
  methods?: any;
  handleAddMethodClick?: any;
  methodsCount?: number;
  activeMethod?: number;
  setActiveMethod?: (id: number) => void;
  setMethods?: any;
  setMethodsCount?: any;
  methodsData?: any;
  resetMethods?: () => void;
  lastMethodId?: number;
  setLastMethodId?: (id: number) => void;
  initialActiveMethod?: number;
  initialShowMethods?: boolean;
  setMethodsFromDraft?: any;
  addNewMethodWithData?: any;
}

export const MethodsContext = createContext<IMethodsContext>({});
