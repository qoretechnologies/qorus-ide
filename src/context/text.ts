import { createContext } from 'react';
import { TTranslator } from '../App';

// Create the text context which will
// store the text function
export const TextContext = createContext<TTranslator | undefined>((t) => t);
