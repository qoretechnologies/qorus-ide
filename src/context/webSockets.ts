import { createContext } from 'react';

export const WebSocketsContext = createContext<WebSocket>(undefined);
