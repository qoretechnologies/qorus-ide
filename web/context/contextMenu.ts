import React, { createContext } from 'react';

export interface IContextMenu {
  event: React.MouseEvent<HTMLElement>;
  data: { item: string, onClick: () => any }[];
  onClose?: () => any;
}

export interface IContextMenuContext {
    addMenu?: (data: IContextMenu) => void;
    removeMenu?: () => void;
}

export const ContextMenuContext = createContext<IContextMenuContext>({});
