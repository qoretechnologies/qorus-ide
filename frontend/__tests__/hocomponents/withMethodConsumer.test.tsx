import React from 'react';
import { create } from 'react-test-renderer';
import { MethodsContext } from '../../src/context/methods';
import withMethods from '../../src/hocomponents/withMethods';

describe('withMethods', () => {
  it('renders the wrapped component with the fields data and functions injected', () => {
    const MockComponent = jest.fn(() => <div />);
    const fieldsData = { data: 'fields data' };

    const tree = create(
      <MethodsContext.Provider value={fieldsData}>
        {React.createElement(withMethods()(MockComponent))}
      </MethodsContext.Provider>
    ).toJSON();

    expect(MockComponent).toHaveBeenCalled();
    expect(tree).toMatchSnapshot();
  });
});
