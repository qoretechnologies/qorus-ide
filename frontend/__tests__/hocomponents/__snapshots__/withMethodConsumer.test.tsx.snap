// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withMethods renders the wrapped component with the fields data and functions injected 1`] = `<div />`;
